lib/jekyll/utils.rb

```diff
@@ -158,5 +158,45 @@ def slugify(string, mode=nil)
         downcase
     end
 
+    # Append an appropriate ending to template so that it matches the specified
+    # permalink style.
+    #
+    # template - permalink template without trailing slash or file extension
+    # permalink_style - permalink style, either built-in or custom
+    #
+    # The returned permalink template will use the same ending style as
+    # specified in permalink_style.  For example, if permalink_style contains a
+    # trailing slash (or is :pretty, which indirectly has a trailing slash),
+    # then so will the returned template.  If permalink_style has a trailing
+    # ":output_ext" (or is :none, :date, or :ordinal) then so will the returned
+    # template.  Otherwise, template will be returned without modification.
+    #
+    # Examples:
+    #   append_permalink_ending("/:basename", :pretty)
+    #   # => "/:basename/"
+    #
+    #   append_permalink_ending("/:basename", :date)
+    #   # => "/:basename:output_ext"
+    #
+    #   append_permalink_ending("/:basename", "/:year/:month/:title/")
+    #   # => "/:basename/"
+    #
+    #   append_permalink_ending("/:basename", "/:year/:month/:title")
+    #   # => "/:basename"
+    #
+    # Returns the updated permalink template
+    def append_permalink_ending(template, permalink_style)
```

Comment:
Yeah, this is a bit fuzzy when just skimming the code. "Ending" is kind of vague and we could shorten `append` to `add`? Maybe `add_permalink_suffix`? Not quite sure what would be best here.